Add new Composable Kernel examples (#397)
amd/rocm-examples: Libraries/ComposableKernel/contraction/batched_contraction/README.md

# CK Tile Programming Model: Batched Tensor Contraction

## Description

This example demonstrates how to perform batched tensor contraction operations

> using the CK Tile programming model. Batched contraction generalizes matrix multiplication to multi-dimensional tensors, supporting arbitrary contraction

Reviewer: ```suggestion
using CK Tile. Batched contraction generalizes matrix multiplication to multi-dimensional tensors, supporting arbitrary contraction
```

patterns across batch, M, N, and K dimensions.

### Supported architectures

> The example is available for

Reviewer: ```suggestion
This example works with
```

[all supported GPU architectures](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html#supported-gpus)

### Application flow

1. Command line arguments are parsed to configure tensor dimensions across G
   (batch), M, N, and K dimensions.
2. Host memory is allocated for input tensors A and B, auxiliary tensors D, and output 
   tensor E.
3. Input tensors are initialized with random values.
4. Device memory is allocated and input data is copied to the device.
5. CK Tile's batched contraction kernel is instantiated and launched on the
   device.
6. If validation is enabled, the results are compared against a CPU reference 
   implementation.
7. Performance metrics including execution time, TFLOPS, and memory bandwidth are 
   reported.
8. All buffers are freed automatically.

## Key APIs and concepts

### CK Tile architecture

> The example makes use of the CK Tile programming model's key components:

Reviewer: This section uses different wording than the previous example's readme (copy). Maybe they should be aligned?

```suggestion
The example makes use of the CK Tile's key components:
```

> * A **shape** defines the hierarchical tile structure and memory layout.

Reviewer: ```suggestion
* The **shape** defines the hierarchical tile structure and memory layout.
```

> * A **problem** combines data types with the shape configuration.

Reviewer: ```suggestion
* The **problem** combines data types with the shape configuration.
```

> * A **pipeline** schedules the sequence of operations for a kernel.

Reviewer: ```suggestion
* The **pipeline** schedules the sequence of operations for a kernel.
```

> * A **kernel** implements the actual computation using the problem and pipeline definitions.

Reviewer: ```suggestion
* The **kernel** implements the actual computation using the problem and pipeline definitions.
```
